dbms/src/Core/AccurateComparison.h

```diff
@@ -41,8 +41,8 @@ using bool_if_not_safe_conversion = std::enable_if_t<!is_safe_conversion<A, B>::
 /// Case 2. Are params IntXX and UIntYY ?
 template <typename TInt, typename TUInt>
 using is_any_int_vs_uint = std::integral_constant<bool,
-                            std::is_integral<TInt>::value && std::is_integral<TUInt>::value &&
-                               std::is_signed<TInt>::value && std::is_unsigned<TUInt>::value>;
+                            std::is_integral<TInt>::value && (std::is_integral<TUInt>::value || std::is_same<TUInt, DB::UInt128>::value) &&
+                               std::is_signed<TInt>::value && (std::is_unsigned<TUInt>::value || std::is_same<TUInt, DB::UInt128>::value)>;
```

Comment:
You have already specialized `std::is_unsinged`.
(What about specializing `std::is_integral`?)